efi/secureboot_policy.go

```diff
@@ -879,7 +879,14 @@ func (g *secureBootPolicyGen) processOSLoadEvent(branches []*secureBootPolicyGen
 		return xerrors.Errorf("cannot determine image type: %w", err)
 	}
 
-	if err := g.computeAndExtendVerificationMeasurement(branches, r, event.Source); err != nil {
+	var source ImageLoadEventSource
+	params := event.params().Resolve(new(loadParams))
```

Comment:
are there tests where NewImageLoadEvent is passed loadParams in and this matters? I don't see new tests in the test file for this file but maybe I miss the new tests/relevant changes

Reply:
I'm not sure I understand the question - there are quite a few tests that pass parameters to NewImageLoadEvent, eg, all of the ones in efi/secureboot_policy_test.go, such as https://github.com/snapcore/secboot/pull/238/files#diff-b20e11ef97bb6e34236a8af13ffa1af982feea79b1317d8c300169c4c123682aR305

I probably should make clear in the code but this specific use of loadParams here is temporary - the new code determines the appropriate behaviour here automatically.

Reply:
sorry it seems I didn't look close enough but they are all using ImageLoadEvenSource?  are you saying that this call to Resolve will go away, it's the only call to Resolve so far? or the usage of ImageLoadEvenSource (that one is commented as going away)

Reply:
Yes, the usage of `ImageLoadEventSource` will be going away in a subsequent PR, but also this call to `Resolve` as this code is going to be replaced as well - there will be another call to `Resolve` that will use a `loadParams` as input which provides inherited parameters for the current node in the profile generation. It's probably easier for me to explain it if I push the current WIP code somewhere.